### PR TITLE
Add example for using event callback function

### PR DESCRIPTION
**Pull request by andreas-bok-sociomantic**
Shows how to use the drizzle_event_watch_fn callback function

### CODE REVIEW

**Comment by ben-palmer-sociomantic** [comment]
One thing I wondered is whether it would be useful to have the expected output as part of the example. It could also print something from the context.

**Comment by andreas-bok-sociomantic** [comment]
Updated based on comments from ben 

```diff 
diff --git a/tests/unit/event_callback.c b/tests/unit/event_callback.c
index c2c9dba..9e9c6db 100644
--- a/tests/unit/event_callback.c
+++ b/tests/unit/event_callback.c
@@ -65,14 +65,18 @@ extern drizzle_return_t drizzle_event_callback(drizzle_st *con, short events,
   void *context)
 {
   ASSERT_TRUE((events != -1));
-  int* cxt_ptr = (int*) drizzle_context(con);
-  ASSERT_NOT_NULL_(cxt_ptr, "Drizzle connection context is null");
-  ASSERT_EQ_(1, cxt_ptr[0], "Invalid drizzle context: Got '%d', Expected '1'",
-    cxt_ptr[0]);
+  char* char_ptr = (char*) drizzle_context(con);
+  ASSERT_NOT_NULL_(char_ptr, "Drizzle connection context is null");
+  ASSERT_STREQ_("Context is everything", char_ptr,
+    "Invalid drizzle context: Got '%s', Expected 'Context is everything'",
+    char_ptr);
+  printf("%s\n", char_ptr);
 
-  cxt_ptr = (int*)context;
-  ASSERT_NOT_NULL_(cxt_ptr, "Drizzle Event Callback context is null");
+  int* cxt_ptr = cxt_ptr = (int*)context;
+
+  ASSERT_NOT_NULL_(cxt_ptr, "Drizzle event callback context is null");
   (*cxt_ptr)++;
+  printf("Drizzle event callback number %d\n", cxt_ptr[0]);
 
   return DRIZZLE_RETURN_OK;
 }
@@ -82,7 +86,7 @@ int main(int argc, char *argv[])
   (void) argc;
   (void) argv;
 
-  int cxt_a = 1;
+  char cxt_a[] = "Context is everything";
   int cxt_b = 0;
 
   drizzle_st *con= drizzle_create(getenv("MYSQL_SERVER"),
@@ -111,5 +115,7 @@ int main(int argc, char *argv[])
   ASSERT_EQ_(cxt_b > 0, true,  "Unexpected number of event callbacks, Got '%d', "
     "Expected '1' or more", cxt_b);
 
+  printf("\nEvent callback was invoked %d times\n", cxt_b);
+
   return EXIT_SUCCESS;
 }

```

File: tests/unit/event_callback.c [hunk 1]
```diff
@@ -65,14 +65,18 @@ extern drizzle_return_t drizzle_event_callback(drizzle_st *con, short events,
...
+  char* char_ptr = (char*) drizzle_context(con);
+  ASSERT_NOT_NULL_(char_ptr, "Drizzle connection context is null");
+  ASSERT_STREQ_("Context is everything", char_ptr,
+    "Invalid drizzle context: Got '%s', Expected 'Context is everything'",
+    char_ptr);
+  printf("%s\n", char_ptr);
 
-  cxt_ptr = (int*)context;
-  ASSERT_NOT_NULL_(cxt_ptr, "Drizzle Event Callback context is null");
+  int* cxt_ptr = cxt_ptr = (int*)context;
```
Comment (ben-palmer-sociomantic) [inline tests/unit/event_callback.c]: Duplicate `cxt_ptr`.

**Review by ben-palmer-sociomantic — COMMENTED** [review]
One comment about the code. Also after the code block in the `rst` file could you have an "Expected Output" section as well?

File: tests/unit/event_callback.c (hunk 1 above)
Comment (andreas-bok-sociomantic) [reply tests/unit/event_callback.c]: oops

**Comment by andreas-bok-sociomantic** [comment]
Fixed duplicate `cxt_ptr`

**Comment by ben-palmer-sociomantic** [comment]
LGTM

**Pull request merged**